astropy/units/core.py

```diff
@@ -1512,46 +1512,41 @@ def to_system(self, system):
         Returns
         -------
         units : list of `CompositeUnit`
-            The list is ranked so that units containing only the base
-            units of that system will appear first.
-        """
-        bases = set(system.bases)
-
-        def score(compose):
-            # In case that compose._bases has no elements we return
-            # 'np.inf' as 'score value'.  It does not really matter which
-            # number we would return. This case occurs for instance for
-            # dimensionless quantities:
-            compose_bases = compose.bases
-            if len(compose_bases) == 0:
-                return np.inf
-            else:
-                sum = 0
-                for base in compose_bases:
-                    if base in bases:
-                        sum += 1
+            The units are ordered by simplicity, preferring base units.
 
-                return sum / float(len(compose_bases))
-
-        x = self.decompose(bases=bases)
-        composed = x.compose(units=system)
-        composed = sorted(composed, key=score, reverse=True)
-        return composed
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> (u.N / u.m**2).to_system(u.si)  # preference for simpler units
+        [Unit("Pa"), Unit("N / m2"), Unit("J / m3"), Unit("6.24151e+18 eV / m3")]
```

Comment:
This is beyond the scope of this pull request, but I really wouldn't expect to see `eV` here.

Reply:
Indeed, `eV` should probably be moved to `misc.py` (like `barn` and `bar`) or perhaps to `astrophys` given that we use it often (like `rydberg`)

Reply:
I also find this output unsettling. I'll open a separate PR to move eV to the `astrophys` system, hoping it'll help get this one moving.